Mathlib/CategoryTheory/Closed/Enrichment.lean

```diff
@@ -0,0 +1,119 @@
+/-
+Copyright (c) 2024 Daniel Carranza. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Daniel Carranza
+-/
+import Mathlib.CategoryTheory.Enriched.Basic
+import Mathlib.CategoryTheory.Closed.Monoidal
+
+/-!
+# A closed monoidal category is enriched in itself
+
+From the data of a closed monoidal category V, we define a V-category structure for V
+where the hom-object is given by the internal hom (coming from the closed structure).
+
+The instance is given at the end of the file.
+
+All structure field values are defined separately
+
+In particular, the proofs of associativity and unitality use the following outline:
+  1. Take adjoint transpose on each side of the equality (uncurry_injective)
+  2. Do whatever rewrites/simps are necessary to apply uncurry_curry
+  3. Conclude with simp (+ a possible exact)
+-/
+
+universe u u₁ v w
+
+namespace CategoryTheory
+
+open MonoidalCategory
+
+namespace MonoidalClosed
+
+variable {V : Type u} [Category.{u₁, u} V] [MonoidalCategory V] [MonoidalClosed V]
+
+/-- The V-identity morphism
+  𝟙_ V ⟶ hom(V, v)
+used to equip V with the structure of a V-category -/
+def id (x : V) : 𝟙_ V ⟶ (ihom x).obj x := curry (ρ_ x).hom
+
+/-- The *uncurried* composition morphism
+  x ⊗ (hom(x, y) ⊗ hom(y, z)) ⟶ (x ⊗ hom(x, y)) ⊗ hom(y, z) ⟶ y ⊗ hom(y, z) ⟶ z.
```

Comment:
I think these render slightly better if you use single backticks ``lean code`` around the verbatim lean code.